spdlog/src/formatter/json_formatter.rs

```diff
@@ -0,0 +1,184 @@
+use std::fmt::{self, Write};
+
+use cfg_if::cfg_if;
+use serde::{ser::SerializeStruct, Serialize};
+
+use crate::{
+    formatter::{FmtExtraInfo, Formatter, LOCAL_TIME_CACHER},
+    Error, Record, StringBuf, __EOL,
+};
+
+struct JsonRecord<'a>(&'a Record<'a>);
+
+impl<'a> Serialize for JsonRecord<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let src_loc = self.0.source_location();
+
+        let mut record =
+            serializer.serialize_struct("JsonRecord", if src_loc.is_none() { 4 } else { 5 })?;
+
+        record.serialize_field("level", &self.0.level())?;
+        {
+            let mut local_time_cacher = LOCAL_TIME_CACHER.lock();
+            record.serialize_field::<str>(
+                "time",
+                local_time_cacher
+                    .get(self.0.time())
+                    .full_second_str()
+                    .as_ref(),
+            )?;
```

Comment:
I think for the machine format JSON, a timestamp number is more friendly. How about we rename the field to `timestamp` and then get the unix timestamp from the cacher?